stdlib/http/src/main/java/org/ballerinalang/net/http/compiler/WebSocketResourceValidator.java

```diff
@@ -186,14 +186,20 @@ private static void validateParamDetailsSize(List<BLangSimpleVariable> paramDeta
 
     private static void validateEndpointParameter(BLangFunction resource, DiagnosticLog dlog,
                                                   List<BLangSimpleVariable> paramDetails, boolean isClient) {
-        if (paramDetails == null || paramDetails.isEmpty() ||
-                (!isClient && !WebSocketConstants.FULL_WEBSOCKET_CALLER_NAME.equals(
-                        paramDetails.get(0).type.toString())) ||
-                (isClient && !WebSocketConstants.FULL_WEBSOCKET_CLIENT_NAME.equals(
-                        paramDetails.get(0).type.toString()))) {
-            dlog.logDiagnostic(Diagnostic.Kind.ERROR, resource.pos, INVALID_RESOURCE_SIGNATURE_FOR
-                    + resource.getName().getValue() + RESOURCE_IN_SERVICE +
-                    ": The first parameter should be an endpoint");
+        if (paramDetails != null && !paramDetails.isEmpty()) {
```

Comment:
What happens when `paramDetails` is null or empty? Shouldn't we also handle it here?

Reply:
It is handled in another function and logged there